dialect/sql/builder.go

```diff
@@ -2232,11 +2229,11 @@ func (n Queries) Query() (string, []interface{}) {
 
 // Builder is the base query builder for the sql dsl.
 type Builder struct {
-	bytes.Buffer               // underlying buffer.
-	dialect      string        // configured dialect.
-	args         []interface{} // query parameters.
-	total        int           // total number of parameters in query tree.
-	errs         []error       // errors that added during the query construction.
+	sb      *strings.Builder // underlying builder.
```

Comment:
Why the lazy allocation?

Reply:
I used the reference, because `strings.Builder` can't be [copied by value](https://golang.org/src/strings/builder.go#L33) like `bytes.Buffer`, and the zero-value of `sql.Builder` is used in almost all builders. It may be cleaner to initialize it properly on the different constructors.